substrate/frame/treasury/src/migration.rs

```diff
@@ -0,0 +1,142 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Treasury pallet migrations.
+
+use super::*;
+use alloc::collections::BTreeSet;
+#[cfg(feature = "try-runtime")]
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+use frame_support::{defensive, traits::OnRuntimeUpgrade};
+
+/// The log target for this pallet.
+const LOG_TARGET: &str = "runtime::treasury";
+
+pub mod cleanup_proposals {
+	use super::*;
+
+	/// Migration to cleanup unapproved proposals to return the bonds back to the proposers.
+	/// Proposals storage item can no longer be created and will be removed in the future.
+	///
+	/// ``
+	///
+	/// Example to provide weight info for migration using [`pallet_balances::Config`]:
```

Comment:
write what should `UnreserveWeight` return, what is meant for, example is more nice to have.